src/kolibri_app/taskbar_icon.py

```diff
@@ -0,0 +1,388 @@
+
+                winreg.SetValueEx(key, f"{APP_NAME}_UI", 0, winreg.REG_SZ, startup_cmd)
+                logging.info("Enabled Kolibri UI startup on logon")
+            else:
+                winreg.DeleteValue(key, f"{APP_NAME}_UI")
+                logging.info("Disabled Kolibri UI startup on logon")
+            return True
+    except (FileNotFoundError, OSError) as e:
+        logging.error(f"Failed to modify UI startup setting: {e}")
+        return False
+
+
+def get_service_start_type():
+    """Check the start type of the Kolibri Windows service."""
+    service_name = SERVICE_NAME
+    try:
+        # Configure subprocess to run hidden (prevents console window flash)
+        startupinfo = subprocess.STARTUPINFO()
+        startupinfo.dwFlags |= subprocess.STARTF_USESHOWWINDOW
+        startupinfo.wShowWindow = subprocess.SW_HIDE
+
+        result = subprocess.run(
+            ["sc", "qc", service_name],
+            capture_output=True,
+            text=True,
+            check=True,
+            startupinfo=startupinfo,
+        )
+        for line in result.stdout.splitlines():
+            line = line.strip()
+            if line.startswith("START_TYPE"):
+                if "2   AUTO_START" in line:
+                    return "auto"
+                if "4   DISABLED" in line:
+                    return "disabled"
+        logging.warning("Could not determine service start type from 'sc qc' output.")
+        return "unknown"
+    except subprocess.CalledProcessError:
+        logging.info(f"Service '{service_name}' not found or 'sc' command failed.")
+        return "not_found"
+    except FileNotFoundError:
+        logging.error("Command 'sc' not found.")
+        return "not_found"
+
+
+class KolibriTaskBarIcon(TaskBarIcon):
+    def __init__(self, app):
+        super(KolibriTaskBarIcon, self).__init__()
+        self.app = app
+        self.server_starting_notified = (
+            False  # Track if we've shown the starting notification
+        )
+
+        self.Bind(wx.adv.EVT_TASKBAR_LEFT_DOWN, self.on_left_click)
+
+        self.set_icon(TRAY_ICON_ICO, f"{APP_NAME}")
+
+    def set_icon(self, path, tooltip):
+        """Sets the icon and tooltip for the taskbar icon."""
+
+        if getattr(sys, "frozen", False):
+            # If running as a PyInstaller bundle, use the _MEIPASS directory
+            resource_path = getattr(sys, "_MEIPASS", os.path.dirname(sys.executable))
+        else:
+            # In development, resource_path will be the project's root directory
+            resource_path = os.path.abspath(".")
+
+        final_path = os.path.join(resource_path, path)
+
+        try:
+            icon = wx.Icon(final_path, wx.BITMAP_TYPE_ICO)
+            self.SetIcon(icon, tooltip)
+        except (FileNotFoundError, wx.wxAssertionError, OSError) as e:
+            logging.error(f"Error setting icon from path '{final_path}': {e}")
+
+    def show_notification(self, title, message, timeout=5):
```

Comment:
Can we make 5 seconds a separate constant, so that we won't need to update this in multiple places if required in future.